src/Bicep.Local.Deploy/Extensibility/LocalExtensibilityHandler.cs

```diff
@@ -5,61 +5,70 @@
 using System.Collections.Concurrent;
 using System.Collections.Immutable;
 using System.Diagnostics;
+using System.Net.Http.Formatting;
+using System.Net.Http;
 using System.Threading;
 using System.Threading.Tasks;
+using System.Web;
+using Azure.Deployments.Engine.Host.Azure.ExtensibilityV2.Contract.Models;
 using Azure.Deployments.Extensibility.Contract;
 using Azure.Deployments.Extensibility.Messages;
 using Bicep.Core.Extensions;
 using Bicep.Core.Registry;
 using Bicep.Core.Semantics;
 using Bicep.Core.Semantics.Namespaces;
 using Bicep.Core.TypeSystem.Types;
+using Microsoft.WindowsAzure.ResourceStack.Common.Json;
 using Microsoft.WindowsAzure.ResourceStack.Common.Utilities;
 using IAsyncDisposable = System.IAsyncDisposable;
 
 namespace Bicep.Local.Deploy.Extensibility;
 
 public class LocalExtensibilityHandler : IAsyncDisposable
```

Comment:
What about `ExtensionHostManager`?

Reply:
I'm just thinking out loud here—what if we tweak the architecture? It seems a bit odd to have a manager or registry of extension hosts. What if we reversed the control? Instead of having a manager invoking an extension host, the `LocalExtensionHost` could reference a `LocalExtensionRegistry`. The `LocalExtensionHost` would use the `LocalExtensionRegistry` to resolve an extension based on a given extension name and version. Then, the `LocalExtensionHost` would dispatch the resolved extension to handle resource operations.

Reply:
Created a diagram to illustrate the idea:
![image](https://github.com/Azure/bicep/assets/16367959/e935b325-83de-418b-af7d-0ece8a21cdc4)

Reply:
This shouldn't block the PR, though. Since it's a design change, we may need to discuss it first and implement it later in subsequent PRs.

Reply:
Agreed, feels like there's definitely scope to simplify the design, and I like the above proposal. Working off it, some additional things to think about:
* Lifecycle management of `IExtension` instances - I assume we'd want this to be the responsibility of the `ExtensionRegistry`?
    * Activating extensions that aren't currently running (currently I believe this happens implicitly when an `IExtension` method call happens, but I think it should be more explicitly managed)
    * Deactivating unused extensions
    * Keepalives (not currently implemented, but may need to be to avoid leaving orphaned processes running)
* Resolving extensions by `name` & `version` leaves room for mixups to occur. I think we should key off the file system (cache) path to avoid this. This would need to be represented in the template somehow for the deployment layer to understand it.